### PR TITLE
feat(photo-library): added cordova-plugin-photo-library

### PR DESCRIPTION
**Pull request by viskin**
PhotoLibrary plugin for Cordova/Ionic Applications.
https://github.com/terikon/cordova-plugin-photo-library

### CODE REVIEW

**Comment by ihadeed** [comment]
The decorators should go in `src/plugin.ts`. 

Overloads need to have a `@private` tag above them so it doesn't confuse the doc generator.

```
   /**
  * @private
  */
  static getThumbnailURL(photoId: string, options?: GetThumbnailOptions): Promise<string>;
   /**
  * @private
  */
   static getThumbnailURL(libraryItem: LibraryItem, options?: GetThumbnailOptions): Promise<string>;
  /**
    * Provides means to request URL of thumbnail, with specified size or quality.
    * @param photo {string | LibraryItem} Id of photo, or LibraryItem.
    * @param options {GetThumbnailOptions} Options, like thumbnail size or quality.
    * @return {Promise<string>} Resolves to URL of cdvphotolibrary schema.
    */
   @Cordova({
     successIndex: 1,
     errorIndex: 2
   })
 static getThumbnailURL(photo: string | LibraryItem, options?: GetThumbnailOptions): Promise<string> { return; }
```

File: src/plugins/photo-library.ts [hunk 1]
```diff
@@ -0,0 +1,252 @@
...
+  })
+  static getAlbums(): Promise<AlbumItem[]> { return; }
+
+  /**
+   * Provides means to request URL of thumbnail, with specified size or quality.
+   * @param photo {string | LibraryItem} Id of photo, or LibraryItem.
+   * @param options {GetThumbnailOptions} Options, like thumbnail size or quality.
+   * @return {Promise<string>} Resolves to URL of cdvphotolibrary schema.
+   */
+  static getThumbnailURL(photoId: string, options?: GetThumbnailOptions): Promise<string>;
```
Comment (ihadeed) [inline src/plugins/photo-library.ts]: Overloads need a `@private` tag so they don't confuse the doc generator. Add the function docs above the `@Cordova` decorator.

File: src/plugins/photo-library.ts [hunk 2]
```diff
@@ -0,0 +1,252 @@
...
+  quality?: number;
+}
+
+export interface GetPhotoOptions {
+}
+
+/**
+ * @private
+ */
+function CordovaFiniteObservable(opts: CordovaFiniteObservableOptions = {}) {
```
Comment (ihadeed) [inline src/plugins/photo-library.ts]: Decorators should be placed in `src/plugins/plugin.ts`

**Comment by viskin** [comment]
Fixed the specified issues.

**Comment by viskin** [comment]
Oops, it seems circleci fails with the change in 2.7.0 (It was tested and worked well in 2.5.1).
Fixed this, opening new PR #1132

**Comment by viskin** [comment]
Hi @ihadeed,
After adding `@private` tag to overloads, there's error from update_docs.sh script on circleci:

> Error: Only one of "private" (or its aliases) allowed. There were 2 - doc "getThumbnailURL" (member)  - from file "/home/ubuntu/ionic-native/src/plugins/photo-library.ts"

In my case, there're 2 overloads that I marked both of them `@private`.

How should it be fixed?

**Comment by ihadeed** [comment]
@viskin I personally prefer not using overloads.

Having overloads requires changing the doc generating process. We would need to ask @perrygovier to have a look and see if it's do-able.

**Pull request merged**